rotkehlchen/data_migrations/migrations/migration_22.py

```diff
@@ -33,10 +33,19 @@ def _remove_legacy_coinbase_keys(rotki: 'Rotkehlchen') -> None:
 
             creds_to_remove.append(cred)
 
-        with rotki.data.db.conn.write_ctx() as write_cursor:
-            write_cursor.executemany(
-                'DELETE FROM user_credentials WHERE name=? AND location=?',
-                [(cred.name, cred.location.serialize_for_db()) for cred in creds_to_remove],
+        if len(creds_to_remove) > 0:
+            with rotki.data.db.conn.write_ctx() as write_cursor:
+                write_cursor.executemany(
+                    'DELETE FROM user_credentials WHERE name=? AND location=?',
+                    [(cred.name, cred.location.serialize_for_db()) for cred in creds_to_remove],
+                )
+                exchange_credentials = rotki.data.db.get_exchange_credentials(write_cursor)
+
+            # Reload the connected exchanges since they are already loaded when migrations are run.
+            rotki.exchange_manager.delete_all_exchanges()
```

Comment:
Why do that? Don't reload all of them. Just delete any potential coinbase one that was deleted. If user has 10 exchanges this will make migration slower no?